_source/_posts/2024-02-27-spring-security-pkce.md

```diff
@@ -0,0 +1,458 @@
+
+
+### Create a simple Spring Boot application
+
+With Spring Initializr and curl, create Spring Boot project:
+
+```shell
+curl -v "https://start.spring.io/starter.zip?\
+bootVersion=3.2.3&\
+language=java&\
+packaging=jar&\
+javaVersion=17&\
+type=gradle-project&\
+dependencies=web,okta,thymeleaf&\
+groupId=com.example&\
+artifactId=spring-web&\
+packageName=com.example.demo" \
+--output spring-web.zip
+```
+
+Then extract the content:
+
+```shell
+unzip spring-web.zip -d spring-web
+```
+
+If you inspect the contents of `build.gradle`, you will find the Okta Spring Boot Starter dependency is included. The Okta Spring Boot Starter simplifies the process of adding authentication into your Spring Boot application by auto-configuring the necessary classes and adhering to best practices, eliminating the need for you to do it manually. It leverages the OAuth 2.0 and OpenID Connect protocols for user authentication.
+
+Add the following additional dependencies to `build.gradle`:
+
+```groovy
+// build.gradle
+dependencies {
+    ...
+    implementation 'me.paulschwarz:spring-dotenv:4.0.0'
+    ...
+}    
+```
+
+Create the `index.html` template at `src/main/resources/templates` with the following content:
+
+```html
+<!-- src/main/resources/templates/index.html -->
+<!doctype html>
+<html lang="en">
+<head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Bootstrap demo</title>
+    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-T3c6CoIi6uLrA9TneNEoa7RxnatzjcDSCmG1MXxSR1GAsXEV/Dwwykc2MPK8M2HN" crossorigin="anonymous">
+    <link rel="stylesheet" href="styles.css">
+</head>
+<body>
+
+<th:block th:replace="~{fragments/navigation.html}" />
+
+
+<div class="container my-5">
+    <h1>Hello, world!</h1>
+    <div class="col-lg-8 px-0">
+        <p class="fs-5">You've successfully loaded up the Bootstrap starter example. It includes <a href="https://getbootstrap.com/">Bootstrap 5</a> via the <a href="https://www.jsdelivr.com/package/npm/bootstrap">jsDelivr CDN</a> and includes an additional CSS and JS file for your own code.</p>
+        <p>Feel free to download or copy-and-paste any parts of this example.</p>
+
+        <hr class="col-1 my-4">
+
+        <a href="https://getbootstrap.com" class="btn btn-primary">Read the Bootstrap docs</a>
+        <a href="https://github.com/twbs/examples" class="btn btn-secondary">View on GitHub</a>
+    </div>
+</div>
+
+<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.2/dist/js/bootstrap.bundle.min.js" integrity="sha384-C6RzsynM9kWDrMNeT87bh95OGNyZPhcTNXj1NW7RuBCsyN/o0jlpcV8Qyq46cDfL" crossorigin="anonymous"></script>
+<script src="main.js"></script>
+</body>
+</html>
+```
+
+Create the fragment `navigation.html` at `src/main/resources/templates/fragments`:
```

Comment:
I would suggest getting rid of these and keep things simple with a single HTML like this (https://github.com/oktadev/auth0-spring-boot-passkeys-demo/blob/main/src/main/resources/templates/home.html) I think you can just copy paste that HTML and remove the fragments and stufff

Reply:
Updated to use single html